File: Algorithm/Python/1000+/2966. Divide Array Into Arrays With Max Difference.py
class Solution:
    def divideArray(self, nums: List[int], k: int) -> List[List[int]]:
        nums.sort()
        idx, n = 0, len(nums)
        while idx < n:
            if nums[idx+2] - nums[idx] > k:
                return []
            idx += 3
        return [nums[i:i+3] for i in range(0, n-2, 3)]

File: Algorithm/C++/1000+/2966. Divide Array Into Arrays With Max Difference.cc
#include <vector>

class Solution {
    public:
        std::vector<std::vector<int>> divideArray(std::vector<int>& nums, int k) {
            std::vector<std::vector<int>> result;
            int n = nums.size();
            std::sort(nums.begin(), nums.end());
            for (int i = 0; i < n - 2; i += 3) {
                if (nums[i+2] - nums[i] > k) {
                    return std::vector<std::vector<int>>();
                } else {
                    result.push_back(std::vector<int>{nums[i], nums[i+1], nums[i+2]});
                }
            }
            return result;
        }
    };